src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -987,39 +1035,92 @@ private int RunCore(TextWriter consoleOutput, ErrorLogger? errorLogger, Cancella
             analyzerDriver = null;
             generatorTimingInfo = null;
 
+            DiagnosticBag? analyzerExceptionDiagnostics = null;
+            
             // Print the diagnostics produced during the parsing stage and exit if there were any errors.
             compilation.GetDiagnostics(CompilationStage.Parse, includeEarlierStages: false, diagnostics, cancellationToken);
+
             if (HasUnsuppressableErrors(diagnostics))
             {
-                return;
-            }
+                if (analyzers.IsEmpty)
```

Comment:
Rather than checking for lack of any `DiagnosticAnalyzer` shouldn't we be checking for lack of any `DiagnosticSuppressor`? Essentially I believe we should return here if `!analyzers.Any(static x => x is DiagnosticSuppressor)`